### PR TITLE
Add additional check to Two-Factor Authentication

### LINKED ISSUE #6113: Add check if MFA app is working, before activate MFA
**1) The issue and what you expected to happen**
    On activating Two-Factor Authentication in the user settings, after type and submit the password you see the the QR code for activating two factor. But at this time there is no way back, so if anything doesn't work in your authenticator app you are locked out and can just reset your password.

I expect that the setting is saved after I successfully typed in my newly generated One-Time Password in an additional input field. So we can prove that the user has successfully added the code.

**2) Exact steps to reproduce this issue**
    Go to user settings with an user without activated Two-Factor Authentication, activate Two-Factor Authentication and type in your password.

### PR DESCRIPTION
**Pull request by wendevlin**
⚠️  This improvement contains a change to the flow and structure of the `/me/tfa` endpoints. If you are relying on these endpoints in your project's logic. Please check the updated docs and apply any changes you might need before upgrading Directus.

---

Closes #6113

### CODE REVIEW

File: api/src/controllers/users.ts [hunk 1]
```diff
@@ -317,14 +317,51 @@ router.post(
...
+
+		const authService = new AuthenticationService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+		const otpValid = await authService.verifyOTP(req.accountability.user, req.body.otp, req.body.secret);
+
+		if (otpValid === false) {
+			throw new InvalidPayloadException(`"otp" is invalid`);
+		}
```
Comment (rijkvanzanten) [inline api/src/controllers/users.ts:356]: Should we move this block of logic into the `enableTFA` method of the users service? It has to be duplicated between the different interfaces (eg REST / GraphQL) now 🤔

Comment (masterwendu) [reply api/src/controllers/users.ts:356]: Yes makes sense. I can change it tomorrow.

Comment (masterwendu) [reply api/src/controllers/users.ts]: I updated the code, so that the logic is within the enableTFA

**Comment by masterwendu** [comment]
FYI, in the view I added to the dialogs form tags. So that you can use Enter to submit password and code.

**Comment by rijkvanzanten** [comment]
Perfect!

**Pull request closed without merging**